R/statistical.R

```diff
@@ -357,6 +357,28 @@ correlation <- function(x, method = "pearson", p.adjust = "none") {
             ncol = ncol(cor_mat$p.value), nrow = nrow(cor_mat$p.value), 
             byrow = TRUE)
     }
+    
+    ## for correlation based on graphical Gaussian models (ggm)
+    if (method == "ggm") {
+        # remove NaN from dataset
+        x <- na.omit(x)
```

Comment:
I was anyway thinking to exclude this step. 
So if it is included it will throw away the features that have a `NA`'s in it, which might be important as you said, and the user will not know. However, the function of `GeneNet` fails if there are any `NA`s in the dataset. But this was also the case for Pearson/ Spearman partial correlation. 
What is your opinion. Should we include it (also for partial correlation) or should this be done by the user manually (so that he is really aware that these will be discarded)?

Reply:
In that case, I would second for keeping the `NA` values and not changing the dimensions for the `statistical` `AdjancencyMatrix` assays. Would it be possible to create an `x_omit <- na.omit(x)` object, run `GeneNet` on it and assigning to a `NA`-containing matrix with dimensions `(nrow(x), nrow(x)` the values of `GeneNet(x_omit`). 

We would then need to make sure that `threshold` and `combine` works on the `NA`-containing adjacency matrix. We can also have an argument in `threshold` that creates `NA` values whenever there is an `NA` value or omits the `NA` values (say we have filter of `pearson_coef > 0.8 and `ggm_coef` > 0.7 - if we have a cell within the `pearson_coef` matrix of 0.8 and within `ggm_coef` of `NA` we will set in the thresholded to 1 in case we omit `NA` values). What do you think about that?

Reply:
We can do it like that. But then we need to implement this` x_omit <- na.omit(x)` etc. also on partial correlation (because it also requires no `NA`s in the data) right? 
I am just not so sure about the last point of thresholding the data. Because if we want to filter the data to be `pearson_coef > 0.8` and` ggm_coef> 0.7` and one row contains a `pearson_coef `of for example `0.99 `and a `ggm_coef` of `NA`, what happens? 
1 ) Will the result be a `1`? --> Then the user gets the impression that this row has a `pearson_coef > 0.8` **and**` ggm_coef> 0.7`
OR
2) Will the result be a `0`/ `NA `--> we lose the information of the `pearson_coef ` that was 0.99 just because `ggm_coef` was `NA`

(But I think option 2 makes still the most sense because of the filter input that is `pearson_coef > 0.8` **and**` ggm_coef> 0.7`)